mailmerge.py

```diff
@@ -136,36 +136,110 @@ def get_merge_fields(self, parts=None):
+            lsecRoot = lastSection.getparent()
+            lsecRoot.remove(lastSection)
+
+            #COPY CHILD ELEMENTS OF BODY IN A LIST
+            childrenList = root.findall('w:body/*', namespaces=NAMESPACES)
+
+            #DELETE ALL SUBELEMENTS OF BODY
             for child in root:
-                root.remove(child)
-                children.append(child)
+                if child.tag == '{%(w)s}body' % NAMESPACES:
+                    child.clear()
 
+            #REFILL BODY AND MERGE DOCS - ADD LAST SECTION ENCAPSULATED OR NOT
+            lr = len(replacements)
+            lc = len(childrenList)
+            parts = []
             for i, repl in enumerate(replacements):
-                # Add page break in between replacements
-                if i > 0:
-                    pagebreak = Element('{%(w)s}br' % NAMESPACES)
-                    pagebreak.attrib['{%(w)s}type' % NAMESPACES] = 'page'
-                    root.append(pagebreak)
-
-                parts = []
-                for child in children:
-                    child_copy = deepcopy(child)
-                    root.append(child_copy)
-                    parts.append(child_copy)
-                self.merge(parts, **repl)
+                for (j, n) in enumerate(childrenList):
+                    element = deepcopy(n)
+                    for child in root:
+                        if child.tag == '{%(w)s}body' % NAMESPACES:
+                            child.append(element)
+                            parts.append(element)
+                            if (j + 1) == lc:
+                                if (i + 1) == lr:
+                                    child.append(mainSection)
+                                    parts.append(mainSection)
+
+                                else:
+
+                                    if separator == 'section':
+                                        intSection = deepcopy(mainSection)
+                                        p   = etree.SubElement(child, '{%(w)s}p'  % NAMESPACES)
+                                        pPr = etree.SubElement(p, '{%(w)s}pPr'  % NAMESPACES)
+                                        pPr.append(intSection)
+                                        parts.append(p)
+                                    elif separator == 'break':
+                                        pb   = etree.SubElement(child, '{%(w)s}p'  % NAMESPACES)
+                                        r = etree.SubElement(pb, '{%(w)s}r'  % NAMESPACES)
+                                        nbreak = Element('{%(w)s}br' % NAMESPACES)
+                                        nbreak.attrib['{%(w)s}type' % NAMESPACES] = type
+                                        r.append(nbreak)
+
+                    self.merge(parts, **repl)
+
+    def merge_pages(self, replacements):
+         """
+         Deprecated method.
+         """
+         print('Please note that this method is deprecated. Use merge_templates method instead')         
```

Comment:
Don't print, but use the logging framework instead, e.g.:

```python
warnings.warn("merge_pages has been deprecated in favour of merge_templates",
                      category=DeprecationWarning,
                      stacklevel=2)
```

Reply:
I agree for combining separators, it's a better way to do. Something like this ? 

 `valid_separators = {'page_break', 'column_break', 'textWrapping_break', 'continuous_section', 'evenPage_section', 'nextColumn_section', 'nextPage_section', 'oddPage_section'}`
`if not separator in valid_separators:`
`    raise ValueError("Invalid separator argument")`
`sepClass, type = separator.split("_")`

For the two existing tests that are still failing (related to merges_pages), it is an expected tree issue.
Before PR a modified test file, I wonder if it would not be a better idea to leave the merge_pages method unchanged at first.
This would allow the user to continue using it in case of an unexpected bug in the merge_templates method.
After a test period, the merge_pages method would be deprecated.
What do you think ?

Reply:
it is actually : 
`type, sepClass = separator.split("_")`